src/main/java/httpServer/RequestParser.java

```diff
@@ -0,0 +1,90 @@
+package httpServer;
+
+import httpServer.http.StatusCodes;
+import httpServer.http.request.HTTPRequest;
+import httpServer.http.request.HTTPRequestBuilder;
+import httpServer.http.response.HTTPResponse;
+import httpServer.outputManagement.ClientWriteable;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.net.Socket;
+
+import static httpServer.http.Constants.CRLF;
+
+public class RequestParser {
+
+    public boolean isRequestLine(String line) {
+        if (line.contains(": ")) {
+            return false;
+        }
+        return true;
+    }
+
+    public boolean atEndOfHeaders(StringBuilder processedInput) {
+        return processedInput.toString().contains(CRLF + CRLF);
+    }
+
+    public HTTPRequest storeHttpRequestInHttpRequestObject(ClientReadable bufferedReader)  {
+
+        HTTPRequestBuilder requestBuilder = new HTTPRequestBuilder();
+        requestBuilder = readStatusCodeAndHeaders(bufferedReader, requestBuilder);
+        int bodyLength = requestBuilder.getContentLengthInt();
+        StringBuilder body = readBody(bufferedReader, bodyLength);
+        return returnHttpRequest(body, requestBuilder);
+
+    }
+
+    private HTTPRequestBuilder readStatusCodeAndHeaders(ClientReadable bufferedReader, HTTPRequestBuilder requestBuilder) {
```

Comment:
Status codes are in responses, not requests. :) Maybe call this `readMetadata` and revise variables below accordingly?